src/toml.cr

```diff
@@ -7,8 +7,12 @@ module TOML
   alias Table = Hash(String, Type)
 
   # Parses a string.
-  def self.parse(string) : TOML::Table
-    Parser.parse(string)
+  def self.parse(input : String | IO) : TOML::Table
+    if input.is_a? String
+      Parser.parse(IO::Memory.new(input))
+    else
+      Parser.parse(input)
+    end
```

Comment:
IMO this is more readable, and maybe an insignificant amount of time faster unless the compiler optimize things up.

```suggestion
  def self.parse(input : String) : TOML::Table
    parse(IO::Memory.new(input))
  end
  
  def self.parse(input : IO) : TOML::Table
    Parser.parse(input)
```

Reply:
Agreed, updated.